pkg/kubelet/nodestatus/setters_test.go

```diff
@@ -443,6 +443,74 @@ func TestNodeAddress(t *testing.T) {
 	}
 }
 
+// We can't test failure or autodetection cases here because the relevant code isn't mockable
+func TestNodeAddress_BareMetal(t *testing.T) {
```

Comment:
nit: `TestNodeAddress_WithoutCloudProvider` might be a bit more accurate naming for this test since bare metal can technically have a "cloud provider" enabled